backend/logic/deploy/build.js

```diff
@@ -0,0 +1,95 @@
+const fs = require('fs')
+const { execFile } = require('child_process')
+
+const { getLogger } = require('../../utils/logger')
+
+const log = getLogger('logic.deploy.build')
+
+/**
+ * Copy all needed files for a shop deployment to the public build directory and
+ * replace HTML template vars to prepare for deployment.
+ *
+ * @param args {object}
+ * @param args.network {object} - Network model instance
+ * @param args.networkConfig {object} - Decrypted network configuration
+ * @param args.shop {object} - Shop model instance
+ * @param args.OutputDir {string} - FS location where the shop build output should go
+ * @param args.dataDir {string} - FS location of the datadir for the shop
+ */
+async function assembleBuild({
+  network,
+  networkConfig,
+  shop,
+  OutputDir,
+  dataDir
+}) {
+  log.info(`Shop ${shop.id}: Preparing data for deploy...`)
+  log.info(`Outputting to ${OutputDir}/public`)
+  await new Promise((resolve, reject) => {
+    execFile('rm', ['-rf', `${OutputDir}/public`], (error, stdout) => {
+      if (error) reject(error)
+      else resolve(stdout)
+    })
+  })
+
+  let publicShopConfig = {}
+  try {
+    const raw = fs.readFileSync(`${OutputDir}/data/config.json`)
+    publicShopConfig = JSON.parse(raw.toString())
+  } catch (e) {
+    log.warning(`Shop ${shop.id}: failed parsing ${OutputDir}/data/config.json`)
+    // TODO: Under which circumstances would it be ok for this to not be a hard error?
```

Comment:
yeah. I think we should throw. this should never happen

Reply:
This was originally in `utils/deployShop`.  Do you know why this was originally a warning?

Reply:
https://github.com/OriginProtocol/dshop/blob/master/backend/utils/deployShop.js#L163

Reply:
yeah. the code was eating that exception but that did not make any sense to me so I added the warning + TODO.
can you maybe check the warning does not show up in the logs? And if not let's throw an exception.

Reply:
There doesn't appear to have been any occurrences of this in the logs, at least within the last 6 months.  Will remove the catch here.

Reply:
Actually, just going to throw a more descriptive error.